docs/software/media.md

```diff
@@ -1124,4 +1126,123 @@ A FOSS web interface media streaming server, including live TV, forked from Emby
+    1) Install CAVA from the DietPi software list. 
+    OR
+    2) Add the following to `/etc/mpd.conf` and restart with `systemctl restart mpd`
+
+    ```sh
+    audio_output {
+        type "fifo"
+        enabled "yes"
+        name "snapcast"
+        path "/tmp/mpd.fifo"
+        format "48000:16:2"
+    }    
+    ```
+    
+    Once you have done that you then need to add the following to `/etc/snapserver.conf` under `[stream]`. The `name` is the name as it will appear to snapcast clients here i've called it MyMpd. Check the snapserver docs for additional parameters you can pass in: <https://github.com/badaix/snapcast/blob/master/doc/configuration.md#pipe>
+
+    ```sh
+    source= pipe:///tmp/mpd.fifo?name=MyMpd&mode=read
+    ```
+
+=== "Using mopidy as an input source"
+    
+    If you have mopidy installed you can use it as an input source.
+    
+    First you need to ensure that mopidy outputs to a pipe. Add the following to `/etc/mopidy/mopidy.conf` and restart with `systemctl restart mopidy`
+
+    ```sh
+    [audio]
+    output = audioresample ! audioconvert ! audio/x-raw,rate=48000,channels=2,format=S16LE ! wavenc ! filesink location=/tmp/mopidy.fifo 
+    ```
+    
+    Once you have done that you then need to add the following to `/etc/snapserver.conf` under `[stream]`. The `name` is the name as it will appear to snapcast clients here i've called it MyMopidy. Check the snapserver docs for additional parameters you can pass in: <https://github.com/badaix/snapcast/blob/master/doc/configuration.md#pipe>
+
+    ```sh
+    source= pipe:///tmp/mopidy.fifo?name=MyMopidy&mode=read
+    ```
+
+=== "Using Raspotify/librespot as an input source"
+    
+    If you have raspotify installed you can use it as an input source.    
+
+    Add the following config under `[stream]`. The `name` is the name as it will appear to snapcast clients here i've called it MySpotify. The `devicename` is the name that will be shown when connecting in Spotify. Check the snapserver docs for additional parameters you can pass in: <https://github.com/badaix/snapcast/blob/master/doc/configuration.md#librespot>. I've disabled the audio cache to protect the SD card.
+
+    ```sh
+    source= librespot:///usr/bin/librespot?name=MySpotify&devicename=SnapcastSpotify&disable_audio_cache=true
+    ```
+
+=== "Using Airport (shareport-sync) as an input
+
+    First install shareport-sync.
+
+    Then (at the moment- see: https://github.com/MichaIng/DietPi/issues/4470) we need to replace the binary in shairport-sync with one that supports stdout. To do this we need to compile a new binary - you will find details here: 
+    <https://github.com/mikebrady/shairport-sync/blob/master/INSTALL.md#build-and-install>
+
+    You need to use this line when configuring instead of the line from the docs:
+
+    ```sh
+    ./configure --sysconfdir=/etc --with-alsa --with-soxr --with-avahi --with-ssl=openssl --with-systemd --with-metadata --with-stdout   
```

Comment:
Actually, when Shairport Sync is only required for this purpose, a minimal build without `--with-alsa` and `--with-systemd` could be done. Not sure if the Snapcast server can deal with metadata, but probably it can be skipped then as well. So the build dependencies and resulting binary size and hence memory usage at least can be reduced a bit 🙂.

Reply:
Snapcast moans a lot without the meta data flag so it is needed. Although I don't yet see where snapcast uses it.

Reply:
I add a little bit about how to disable the original service or that you can try to use different ports.

Reply:
Disabling the service is sufficient, this prevents systemd from starting it, but dietpi-postboot>dietpi-services will still start it. Either it needs to be excluded: `dietpi-services` > `shairport-sync` > `Include/Exclude` > `Exclude`
or easier, masked:
```sh
systemctl mask shairport-sync
```